packages/vscode-container-client/src/utils/spawnStreamAsync.ts

```diff
@@ -201,24 +202,66 @@ export class NoShell extends Shell {
     }
 }
 
+/**
+ * Options for spawning a process with pipe streaming capabilities.
+ * This extends the standard {@link SpawnOptions} to include additional features
+ * such as command logging, cancellation support, and shell customization.
+ */
 export type StreamSpawnOptions = SpawnOptions & {
+    /**
+     * A callback that is invoked with the full command line that is being executed.
+     * This is useful for logging or debugging purposes.
+     */
     onCommand?: (command: string) => void;
+
+    /**
+     * A cancellation token that can be used to cancel the command execution.
+     * If the token is cancelled, the command will be terminated and the promise will be rejected
+     */
     cancellationToken?: CancellationTokenLike;
+
+    /**
+     * A shell provider that can be used to customize the shell used for command execution.
+     * This needs to be used in conjunction with {@link SpawnOptions.shell}.
+     */
     shellProvider?: Shell;
 
+    /**
+     * Whether to allow the use of unsafe executable paths. If true, the command executed
+     * could be executed from the current working directory instead of the PATH, which may not be safe.
+     */
+    allowUnsafeExecutablePath?: boolean;
```

Comment:
For customized commands, we may be able to do a regex like `^(['"])[\w\s]*\1|^\w*` to extract the specific command.